### ISSUE TITLE
fix epytext problems in SVN HEAD

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2640]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)| @mwhudson reported|
|-|-|
|Trac ID|trac#2640|
|Type|enhancement|
|Created|2007-05-14 07:42:00Z|

This patch fixes all outright epytext errors in SVN HEAD (i.e. things that make epydoc give up, not things that make the output look terrible).

It was created in-browser using my online docstring editing thingy, which may be approaching usefulness now...

Attachments:

* [twisted-epytext-fixes.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3a2/3a2722c81aa17a62d13afa60bd96639c7f2caa65/f5b42c010d16db6bdb409f24e02cc9a5b5e42e45.diff) (10167 bytes) - added by mwh on 2007-05-14 07:42:20Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2640 2640
type__enhancement enhancement
reporter__mwh mwh
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1179128520000000 1179128520000000
changetime__1179701447000000 1179701447000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set owner to @mwhudson|
|-|-|

Can we import some kind of trial test from pydoctor to prevent this problem from recurring?
```
<exarkun> the patch fixes machine-verifiable defects in the docstrings
<exarkun> therefore, there should be machine-verification for the correctness of that property of the docstrings
```

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_2"></a>|@mwhudson commented|
|-|-|

No, but it should be fairly straightforward to add a function like 

def improperEpytextForPackage(filePath, systemcls="pydoctor.model.System"):
    ...

This would mean that pydoctor depends on nevow which depends on twisted which depends on pydoctor though, which is kind of crazy.  The above function could be implemented in a way that wouldn't require pydoctor to import nevow, but that would be a bit more work.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

A somewhat simpler first step might be to simply run pydoctor over Twisted on the doc buildslave.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_4"></a>|@mwhudson commented|
|-|-|

Indeed.  This should actually be easier now that the last time we talked about this: pydoctor exits with a code of 2 when it has found at least one problem docstring.

Extra brownie points might result from putting the html generated as a side effect of this somewhere publicly accessible, but well, let's not worry about that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Step set up on a new doc-only builder.  Applied the patch to a branch, epytext-fixes-2640, and made a couple further tweaks.

Someone else should take a quick look.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @exarkun|
|-|-|

I did have a look, and it looks cool. I just didn't understand the change in pop3, but that's OK, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [20346]) Merge epytext-fixes-2640

Author: mwh, exarkun
Reviewer: exarkun, therve
Fixes [#2640](https://github.com/twisted/twisted/issues/2640)

Fix mangled epytext markup in docstrings project-wide.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
